catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -302,7 +302,7 @@ private Result handleTake() {
         Result result = null;
         try {
             result = queue.take();
-            if (result == poisonPillResult) {
+            if (result == POISON_PILL_RESULT) {
```

Comment:
All right that sounds good to me. I actually thought we were already overriding equals but we were not... bad assumption. So that probably explains also why `==` was there for so long. It is almost like the code really wanted to check for instanceof POISON_PILL_RESULT without the performance hit since POISON_PILL_RESULT really doesn't have anything it.